### PR TITLE
media-sound/mpdscribble: add version 0.23

### PR DESCRIPTION
**Pull request by ArsenArsen**
Addresses https://bugs.gentoo.org/812275 and https://bugs.gentoo.org/784398

Keywords in `0.23` were adjusted to my testing abilities (and even with that might be inaccurate still, since I haven't explicitly tested on `x86` alone). I believe bug https://bugs.gentoo.org/605374 can also be addressed in this commit, with some further changes.

### CODE REVIEW

File: media-sound/mpdscribble/mpdscribble-0.23.ebuild [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (juippis) [inline media-sound/mpdscribble/mpdscribble-0.23.ebuild:4]: 8 should work now.

File: media-sound/mpdscribble/mpdscribble-0.23.ebuild [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+inherit meson systemd
+
+DESCRIPTION="An MPD client that submits information to Audioscrobbler"
+HOMEPAGE="https://www.musicpd.org/clients/mpdscribble/"
```
Comment (juippis) [inline media-sound/mpdscribble/mpdscribble-0.23.ebuild:9]: Could you also add the Github page as 2ndary HOMEPAGE here?

File: media-sound/mpdscribble/mpdscribble-0.23.ebuild [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="
+	media-libs/libmpdclient
+	net-misc/curl
+	dev-libs/boost
+	dev-libs/libgcrypt
+"
```
Comment (juippis) [inline media-sound/mpdscribble/mpdscribble-0.23.ebuild:21]: Please sort

File: media-sound/mpdscribble/files/mpdscribble-0.23-Don-t-install-AUTHORS-COPYING-NEWS-README.rst.patch [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+-  'AUTHORS', 'COPYING', 'NEWS', 'README.rst',
+-  install_dir: join_paths(get_option('datadir'), 'doc', meson.project_name()),
+-)
+-
+ subdir('systemd')
+ 
+ subdir('doc')
+-- 
+2.32.0
+
```
Comment (juippis) [inline media-sound/mpdscribble/files/mpdscribble-0.23-Don-t-install-AUTHORS-COPYING-NEWS-README.rst.patch:29]: I guess there's some sort of conflict with compressing these files properly?

File: media-sound/mpdscribble/mpdscribble-0.23.ebuild [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+DOCS=( AUTHORS COPYING NEWS README.rst )
+
+DEPEND="${RDEPEND}"
```
Comment (juippis) [inline media-sound/mpdscribble/mpdscribble-0.23.ebuild:25]: Please move this below RDEPEND.

File: media-sound/mpdscribble/mpdscribble-0.23.ebuild [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-0.23-Unconditionally-install-systemd-unit-files.patch
+	"${FILESDIR}"/${PN}-0.23-Don-t-install-AUTHORS-COPYING-NEWS-README.rst.patch
+)
+
+src_prepare() {
+	# we install these below via systemd_do{,user}unit
+	sed -i '/install_dir/d' -i systemd/*/meson.build
```
Comment (juippis) [inline media-sound/mpdscribble/mpdscribble-0.23.ebuild:34]: Don't you already introduce some sort of patch file for this? Why's the sed needed anymore? Besides, external commands (`sed`) needs to `|| die`.

**Review by juippis — COMMENTED** [review]
Okay let's process this. Few notes.

File: media-sound/mpdscribble/mpdscribble-0.23.ebuild [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="
+	media-libs/libmpdclient
+	net-misc/curl
+	dev-libs/boost
+	dev-libs/libgcrypt
```
Comment (juippis) [inline media-sound/mpdscribble/mpdscribble-0.23.ebuild:20]: Also if this gets linked, add a subslot binder, 
`dev-libs/libgcrypt:=`

File: media-sound/mpdscribble/files/mpdscribble-0.23-Don-t-install-AUTHORS-COPYING-NEWS-README.rst.patch (hunk 4 above)
Comment (ArsenArsen) [reply media-sound/mpdscribble/files/mpdscribble-0.23-Don-t-install-AUTHORS-COPYING-NEWS-README.rst.patch:29]: yes, a version with and without compression would be installed

**Comment by ArsenArsen** [comment]
All that should be fixed, I believe.

BTW, is the sort order for fields defined anywhere? For future reference

**Comment by juippis** [comment]
> BTW, is the sort order for fields defined anywhere? For future reference

Theory:
https://devmanual.gentoo.org/general-concepts/dependencies/index.html

Practical example:
https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide/Style_Guide#Sorting_dependencies

FYI getting:
```
 * QA Notice: This ebuild installs into paths that should be created at runtime.
 *  To fix, simply do not install into these directories.  Instead, your package
 *  should create dirs on the fly at runtime as needed via init scripts/etc...
 * 
 *   var/cache
 *   var/cache/mpdscribble
```
Contribution looks fine, but I'd like to hear your opinion on this: Do you think you can modify the systemd service file & openrc init script to accomplish this? It's not a requirement, will merge if you don't want to put time for that now, but it'd be a great extra. Let me know.

**Comment by ArsenArsen** [comment]
I did ask about this on IRC, since this is an old issue. This seems like a job for the *tmpfile implementations. I can add the configuration for those.

**Review by juippis — APPROVED** [review]
Well let's merge and fix the issues this PR was originally raised for. If you do end up fixing the init.d/service files, please ping me directly in Github so I see it (not part of the project that maintains this)

**Pull request closed without merging**